src/main/webapp/app/modules/canvas/filter/filter-element.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { useEffect, useState } from 'react';
```

Comment:
this should contain only html to render..move all functions in filter util service